packages/react/src/config.ts

```diff
@@ -19,4 +20,5 @@ export type LensConfig = {
   logger?: ILogger;
   storage: IStorageProvider | IObservableStorageProvider;
   sources?: Sources;
+  appId?: AppId;
```

Comment:
After this change, should we ensure that if one provides `appId` there is also `sources` where one of the item is the same `appId`?  It could lead to an odd UX if `sources` is configured for example with `X` and `Y` but `appId` publishes on, let's say `Z`.

Reply:
I personally think it should be an invariant check, I cannot think of a normal scenario where one can get this configuration wrong... @desfero what do you think?

Also bear in mind that providing `appId` and not provide `sources` is a legitimate configuration. It means one does want to flag publication as originated from `appId` but it wants to show data from the entire ecosystem (🤔 ).

I would fail as soon as possible. A while back I was entertaining the idea if we need some sort of config helper like wagmi does.

The helper could accept raw JS types and return properly branded `sources` or `appId` for example. Over time it might be a place to do validation (actually the check we are talking about seems validation indeed). Thoughts?

Reply:
I added `validateConfig` helper that is called in `LensProvider`, this way I can test it independently.
Regarding config helper, I think it would be an overkill for now, I understand you have sth like the below on your mind.

```
const { sources, appId } = buildConfig('my-app-id')

const lensConfig = {
  bindings: wagmiBindings(),
  environment: staging,
  storage: localStorage(),
  sources, 
  appId
}

```

Reply:
I was thinking more akin to wagmi `createClient`, so in our case:
```ts

import { createConfig } from '@lens-protocol/react`;

const config = createConfig({
  source: ['foo', 'bar'], // from a type perspective this is Array<string>
  appId: 'foo' // string
  ...
})
```
The generated `config`  can then be opaque to the consumer but for the purpose of this conversation we can assume:
```ts
config.appId // AppId
condif.sources // Array<AppId> OR Sources which is the same
```



```

Reply:
maybe let's wait and see how the actual appId format/generation will look like and then we will decide if a helper is needed?